src/Microsoft.OData.Edm/Csdl/Serialization/EdmModelCsdlSchemaXmlWriter.cs

```diff
@@ -826,14 +832,28 @@ internal override void WriteNavigationPropertyBinding(IEdmNavigationPropertyBind
             }
         }
 
-        private static string SridAsXml(int? i)
+        private string SridAsXml(int? i)
         {
-            return i.HasValue ? Convert.ToString(i.Value, CultureInfo.InvariantCulture) : CsdlConstants.Value_SridVariable;
+            if ((this.writerSettings?.LibraryCompatibility & EdmLibraryCompatibility.UseLegacyVariableCasing) == EdmLibraryCompatibility.UseLegacyVariableCasing)

```

Comment:
`this.writerSettings?.LibraryCompatiblity` = `this.writerSettings.LibraryCompatiblity` if `this.writerSettings` is not expected to be null